sdk/servicebus/service-bus/src/sender.ts

```diff
@@ -230,12 +238,32 @@ export class SenderImpl implements Sender {
     );
     throwTypeErrorIfParameterMissing(this._context.namespace.connectionId, "message", message);
 
+    const retryOptions = this._senderOptions.retryOptions || {};
+    retryOptions.timeoutInMs = getRetryAttemptTimeoutInMs(retryOptions);
+
     const messages = [message];
-    const result = await this._context.managementClient!.scheduleMessages(
-      scheduledEnqueueTimeUtc,
-      messages
-    );
-    return result[0];
+
+    const scheduleMessageOperationPromise = () =>
+      new Promise<Long.Long>(async (resolve, reject) => {
+        try {
+          const result = await this._context.managementClient!.scheduleMessages(
+            scheduledEnqueueTimeUtc,
+            messages,
+            retryOptions.timeoutInMs!
+          );
+          resolve(result[0]);
+        } catch (error) {
+          reject(error);
+        }
+      });
```

Comment:
Just curious... can the above not be simplified like this:

```javascript
const scheduleMessageOperationPromise = async () => {
  const result = await this._context.managementClient!.scheduleMessages(
             scheduledEnqueueTimeUtc,
             messages,
             retryOptions.timeoutInMs!
           );
      return result[0];
}
```

Reply:
Same for all the other places that uses the same pattern

Reply:
I thought I was just following the convention and say.. in case we want to process the error for logging or otherwise.

Updated all the places similar to your snippet now.